content/en/docs/ops/configuration/traffic-management/dns-proxy/index.md

```diff
@@ -0,0 +1,111 @@
+
+While Kubernetes provides DNS resolution for Kubernetes `Service`s out of the box, any custom `ServiceEnty`s. With this feature, `ServiceEnty` addresses can be resolved without requiring custom configuration of a DNS server. For Kubernetes `Service`s, the DNS response will be the same, but with reduced load on `kube-dns` and increased performance.
+
+This functionality is also available for services running outside of Kubernetes. This means that all internal services can be resolved without clunky workarounds to expose Kubernetes DNS entries outside of the cluster.
+
+## Getting Started
+
+This feature is not currently enabled by default. To enable it, install Istio with the following settings:
+
+{{< text bash >}}
+$ cat <<EOF | istioctl install -y -f -
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+spec:
+  meshConfig:
+    defaultConfig:
+      proxyMetadata:
+        # Enable basic DNS proxying
+        ISTIO_META_DNS_CAPTURE: "true"
+        # Enable automatic address allocation, optional
+        ISTIO_META_DNS_AUTO_ALLOCATE: "true"
+EOF
+{{< /text >}}
+
+This can also be enabled on a per-pod basis with the [`proxy.istio.io/config`annotation ](/docs/reference/config/annotations/).
+
+When deploying a VM using [`istioctl workload entry configure`](/docs/setup/install/virtual-machine/), basic DNS proxying will be enabled by default.
+
+## DNS Capture In Action
+
+To try out the DNS capture, first setup a `ServiceEntry` for some external service.
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1beta1
+kind: ServiceEntry
+metadata:
+  name: external-address
+spec:
+  addresses:
+  - 198.51.100.0
+  hosts:
+  - address.internal
+  ports:
+  - name: http
+    number: 80
+    protocol: HTTP
+{{< /text >}}
+
+Without the DNS capture, a request to `address.internal` would likely fail to resolve. Once this is enabled, we should instead get a response back based on the configured `address`:
+
+{{< text bash >}}
+$ curl -v address.internal
+*   Trying 198.51.100.1:80...
+{{< /text >}}
+
+## Address Auto Allocation
+
+In the example above, we had a predefined IP address for the service that we wanted to send. However, its common to access external services which do not have stable addresses, and instead rely on DNS. In this case, the DNS proxy will not have enough information to return a response, and will need to forward DNS requests upstream.
+
+This is especially problematic with TCP traffic. Unlike HTTP requests, which are routed based on `Host` headers, TCP carries much less information; we can only route on the destination IP and port number. Because we do not have a stable IP for the backend, we cannot route based on that either, leaving only port number, which leads to conflicts when multiple `ServiceEntry`s for TCP services share the same port.
```

Comment:
```suggestion
This is especially problematic with TCP traffic. Unlike HTTP requests, which are routed based on `Host` headers, TCP carries much less information; you can only route on the destination IP and port number. Because you don't have a stable IP for the backend, you cannot route based on that either, leaving only port number, which leads to conflicts when multiple `ServiceEntry`s for TCP services share the same port.
```